src/libtomahawk/CMakeLists.txt

```diff
@@ -533,8 +533,7 @@ TARGET_LINK_LIBRARIES(${TOMAHAWK_LIBRARY} PRIVATE
 )
 
 # TagLib
-TARGET_LINK_LIBRARIES(${TOMAHAWK_LIBRARY} LINK_PUBLIC ${TAGLIB_LIBRARIES})
-TARGET_INCLUDE_DIRECTORIES(${TOMAHAWK_LIBRARY} PUBLIC ${TAGLIB_INCLUDES})
+target_link_libraries(${TOMAHAWK_LIBRARY} LINK_PUBLIC Taglib::Taglib)
```

Comment:
This will no longer work if taglib was compiled and installed locally with zlib enabled.

Reply:
Perfectly possible I was missing something there. Just re-add `target_include_directories`?

Reply:
Its actually the ECM module that does not read out the pkg config correctly and will in turn not append `-lz` to the `${Taglib_LIBRARIES}` variable.

I think a simple work around here would be 

```
IF(ZLIB_FOUND)
    target_link_libraries(${TOMAHAWK_LIBRARY} PUBLIC -lz)
endif()
```

Reply:
That would be a bug in taglib though and fixed in master: https://github.com/taglib/taglib/commit/ef1312d62239f399c40233d76ef3328b8dadf984

Unsure how best to proceed there. We can't depend on a yet-unreleased version of taglib, though basically every distribution has that covered by now by shipping a snapshot of taglib instead. https://repology.org/project/taglib/versions

Reply:
Here's what Clementine proj are doing: https://github.com/clementine-player/Clementine/blob/master/CMakeLists.txt#L127

Although it is a desperate move, as taglib upstream have not bumped that version yet from `1.11.1`. And that leads to us patching our taglib snapshot to be `1.11.2`...

Reply:
How about something like:

```
if (NOT TAGLIB_VERSION VERSION_GREATER 1.11.1)
  IF(ZLIB_FOUND)
    target_link_libraries(${TOMAHAWK_LIBRARY} PUBLIC -lz)
  endif()
endif()
```

Reply:
Thanks, that seems like a sane way forward :)